scripts/mass_user_emails.py

```diff
@@ -212,12 +258,23 @@ def get_eligible_users(user_detail_email_key, force=False):
             extra_details__isnull=True,
         )
     )
+    for query in USER_EXCLUDE:
+        eligible_users = eligible_users.exclude(**query)
     if not force:
         eligible_users = eligible_users.exclude(
             extra_details__private_data__has_key=user_detail_email_key,
         )
-    return eligible_users.annotate(email_lowercase=Lower('email')).distinct(
-        'email_lowercase'
+    # last step: convert email to lowercase and strip any filters
+    # regex finds 'name+filter@example.com' type addresses
+    return eligible_users.annotate(email_cleaned=Func(
+            Lower(F('email')),
+            Value(r'\+[^*]*@'),
+            Value(''),
+            Value('g'),
+            function='REGEXP_REPLACE',
+            output_field=CharField(),
+    )).distinct(
+        'email_cleaned'
```

Comment:
I think this doesn't quite do what's expected. My reading of the regex replacement is:
1. Find any sequence of a literal `+` ,followed by zero or more characters that are not `*`, followed by a `@`
1. …and replace all that with nothing
1. …and repeat this until no matches remain.

The result is, if I have:
| `email=` | …then I get `email_cleaned=` |
| - | - |
| john@john.john | john@john.john |
| john+clone2@john.john | johnjohn.john |
| john+clone99@john.john | johnjohn.john |

After the `distinct()` call, the emails remaining as recipients are:
* john@john.john and
* either john+clone99@john.john or john+clone2@john.john (these are returned indeterminately by PostgreSQL).

I'd suggest cleaning up the regex and then taking advantage of your annotation to send mail to `user.email_cleaned`; that way, messages always go to `john@john.john` instead of `john+cloneblah@john.john`.

Reply:
Thank you for noticing that! I've updated both points and tested the regex (in context and from the shell).